src/components/select/UISelect.vue

```diff
@@ -331,34 +341,46 @@ export default defineComponent({
 
             selected.value = props.multi ? [] : null;
         };
+
         const isEqual = (previous: Option, next: Option): boolean => {
+            // Compare as strings options (checking type of previous/next to satisfy TS).
+            if (stringOptions.value || typeof previous === 'string' || typeof next === 'string') {
+                const previousString: string = typeof previous === 'string' ? previous : previous[props.optionLabel];
+                const nextString: string = typeof next === 'string' ? next : next[props.optionLabel];
```

Comment:
I think it would be safer if used optionKey as we know that that would defo be unique because that's what we use elsewhere as a unique value

Reply:
Given the object is internally constructed we can use either and be sure of the value (as in, both the label and key will equal the original string). But I can change it to `key` for consistency.

Reply:
> Given the object is internally constructed 

Not every time?

> But I can change it to key for consistency.

I would prefer that for the sake of brevity